shedlock-provider-jedis/src/main/java/net/javacrumbs/shedlock/provider/jedis/JedisLockProvider.java

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.javacrumbs.shedlock.provider.jedis;
+
+import net.javacrumbs.shedlock.core.LockConfiguration;
+import net.javacrumbs.shedlock.core.LockProvider;
+import net.javacrumbs.shedlock.core.SimpleLock;
+import net.javacrumbs.shedlock.support.LockException;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.util.Optional;
+
+/**
+ * Uses Redis's `SET resource-name anystring NX PX max-lock-ms-time` as locking mechanism.
+ *
+ * See https://redis.io/commands/set
+ *
+ */
+public class JedisLockProvider implements LockProvider {
+
+    private static final String keyPrefix = "job-lock";
+
+    private static final String DEFAULT_ENV = "default";
+
+    private JedisPool jedisPool;
+    private String environment;
+
+    public JedisLockProvider(JedisPool jedisPool) {
+        this(jedisPool, DEFAULT_ENV);
+    }
+
+    public JedisLockProvider(JedisPool jedisPool, String environment) {
+        this.jedisPool = jedisPool;
+        this.environment = environment;
+    }
+
+    @Override
+    public Optional<SimpleLock> lock(LockConfiguration lockConfiguration) {
+        long difference = getDifference(lockConfiguration);
+        if (difference > 0) {
+            String key = buildKey(lockConfiguration.getName(), this.environment);
+            try (Jedis jedis = jedisPool.getResource()) {
+                String rez = jedis.set(key, buildValue(), "NX", "PX", difference);
+                if (rez != null && "OK".equals(rez)) {
+                    return Optional.of(new RedisLock(key, jedisPool));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    long getDifference(LockConfiguration lockConfiguration) {
+        long now = Instant.now().toEpochMilli();
+        long mostDiff = lockConfiguration.getLockAtMostUntil().toEpochMilli() - now;
+        long leastDiff = lockConfiguration.getLockAtLeastUntil().toEpochMilli() - now;
+
+        long difference = -1;
+        if (mostDiff > 0 && leastDiff > 0) {
+            difference = Math.max(mostDiff, leastDiff);
+        } else if (mostDiff > 0 && leastDiff <= 0) {
+            difference = mostDiff;
+        } else if (mostDiff <= 0 && leastDiff > 0) {
+            difference = leastDiff;
+        }
+        return difference;
+    }
+
+    private static final class RedisLock implements SimpleLock {
+        private final String key;
+        private final JedisPool jedisPool;
+
+        private RedisLock(String key, JedisPool jedisPool) {
+            this.key = key;
+            this.jedisPool = jedisPool;
+        }
+
+        @Override
+        public void unlock() {
+            try (Jedis jedis = jedisPool.getResource()) {
+                jedis.del(key);
```

Comment:
Just to explain - lockAtLeastFor is a hack used by people with really short tasks that want to prevent double execution.